libs/sdk-core/src/moon_pay/moon_pay_api.rs

```diff
@@ -0,0 +1,23 @@
+use anyhow::Result;
+
+use crate::moon_pay::moon_pay_config::MoonPayConfig;
+use crate::moon_pay::moon_pay_url_signer::MoonPayUrlSigner;
+use crate::SwapInfo;
+
+pub struct MoonPayApi {
+    config: MoonPayConfig,
+    signer: Box<dyn MoonPayUrlSigner>,
+}
+
+impl MoonPayApi {
+    pub fn new(config: MoonPayConfig, signer: Box<dyn MoonPayUrlSigner>) -> Self {
+        Self { config, signer }
+    }
+
+    pub async fn sign_moon_pay_url(&mut self, swap_info: &SwapInfo) -> Result<String> {
```

Comment:
Why do we need mut here? This forced the code to use mutex in BreezServices and I don't see why we need mut reference. If we omit the mut then we wont need the mutex.